bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/StackRenderer.java

```diff
@@ -209,6 +217,40 @@ private void addItemToSet(List<CTabItem> itemsToSet, MPart parentParent) {
 		}
 	}
 
+	@Inject
+	@Optional
+	void subscribePerspectiveSwitched(
+			@UIEventTopic(UIEvents.UILifeCycle.PERSPECTIVE_SWITCHED) org.osgi.service.event.Event event) {
+		Object element = event.getProperty(EventTags.ELEMENT);
+		if (element instanceof MPerspective) {
+			setOnboarding((MPerspective) element);
+		}
+	}
+
+	private void setOnboarding(MPerspective perspective) {
+		if (onBoarding == null) {
+			return;
+		}
+
+		Image image = getImageFromURI(perspective.getIconURI());
```

Comment:
This should be:

```suggestion
		Image image = getImageFromURI(perspective.getIconURI());
		if (image.isDisposed()) {
			return;
		}
```

This at least fixed the failing test locally on my computer